storages/backends/azure_storage.py

```diff
@@ -390,16 +399,9 @@ def list_all(self, path=""):
 
     def listdir(self, path=""):
         """
-        Return directories and files for a given path.
-        Leave the path empty to list the root.
-        Order of dirs and files is undefined.
+        Return all files for a given path.
+        Given that Azure can't return paths it only returns files.
+        Works great for our little adventure.
         """
-        files = []
-        dirs = set()
-        for name in self.list_all(path):
-            n = name[len(path) :]
-            if "/" in n:
-                dirs.add(n.split("/", 1)[0])
-            else:
-                files.append(n)
-        return list(dirs), files
+        
+        return [], self.list_all(path)
```

Comment:
So, while technically true, in practice all of these services use "/" as a directory separator.

What is the issue we are trying to solve here and what is the solution?

Reply:
The issue here is that Django tries to "walk" the directories, so first top level, next folder level and so on.
Azure does not support this type of walk, since they don't support listing directories, only blobs.
The only purpose for Djangos walk is to find all the files, but since we can just find all files with the list_all(), we just return all files and none directories.